### PR TITLE
Create picasso targets for drawableStart and drawableEnd on TextViews

### PR DESCRIPTION
**Pull request by frett**

### CODE REVIEW

File: gto-support-picasso/src/main/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTarget.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    companion object {
+        fun of(view: TextView) = view.getTag(R.id.picasso_textViewDrawableEndTarget) as? TextViewDrawableEndTarget
+                ?: TextViewDrawableEndTarget(view)
+    }
+
+    public override fun updateDrawable(drawable: Drawable?) {
+        val current = view.compoundDrawablesRelative
+        drawable?.apply { setBounds(0, 0, intrinsicWidth, intrinsicHeight) }
+        view.setCompoundDrawablesRelative(drawable, current[1], current[2], current[3])
```
Comment (frett) [inline gto-support-picasso/src/main/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTarget.kt:21]: you are setting `drawable` to the start drawable here, it should replace the end drawable

Comment (TheNoodleMoose) [reply gto-support-picasso/src/main/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTarget.kt:21]: So it needs to be set to ```current``` instead then as that is referencing the end drawable?

File: gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTargetTest.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        target.updateDrawable(drawable)
+        verify(drawable).setBounds(0, 0, WIDTH, HEIGHT)
+        argumentCaptor<Drawable> {
+            verify(view).setCompoundDrawablesRelative(
+                eq(compoundDrawables[0]),
+                eq(compoundDrawables[1]),
+                capture(),
+                eq(compoundDrawables[3])
+            )
+            assertSame(drawable, firstValue)
```
Comment (TheNoodleMoose) [inline gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTargetTest.kt:67]: Does this ```capture``` function create the value ```firstValue``` for the test? I don't see that value being set anywhere else.

**Review by TheNoodleMoose — APPROVED** [review]
Looks good to me!

File: gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTargetTest.kt (hunk 2 above)
Comment (frett) [reply gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTargetTest.kt:67]: yeah, it's the `ArgumentCaptor` feature of Mockito
https://site.mockito.org/javadoc/current/org/mockito/ArgumentCaptor.html

Comment (frett) [reply gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTargetTest.kt:67]: combined with the mockito-kotlin extensions

Comment (frett) [reply gto-support-picasso/src/test/java/org/ccci/gto/android/common/picasso/widget/TextViewDrawableEndTargetTest.kt:67]: https://github.com/mockito/mockito-kotlin/wiki/Mocking-and-verifying#argument-captors

**Review by TheNoodleMoose — APPROVED** [review]

**Pull request merged**